docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,470 @@
+.. _building_the_project_with_dpcpp:
+
+Building the Project with DPC++
+===============================
+
+This page describes building the oneMKL Interfaces with either the Intel(R)
+oneAPI DPC++ Compiler or open-source oneAPI DPC++ Compiler. For guidance on
+building the project with AdaptiveCpp, see
+:ref:`building_the_project_with_adaptivecpp`.
+
+.. _build_setup_with_dpcpp:
+
+Environment Setup
+##################
+
+#. 
+   Install the required DPC++ compiler (Intel(R) DPC++ or Open DPC++ - see
+   :ref:`Selecting a Compiler<selecting_a_compiler>`).
+
+#. 
+   Clone this project. The root directory of the cloned repository will be
+   referred to as ``<path to onemkl>``.
+
+#. 
+   Build and install all `required dependencies
+   <https://github.com/oneapi-src/oneMKL?tab=readme-ov-file#software-requirements>`_. 
+
+.. _build_introduction_with_dpcpp:
+
+Build Commands
+###############
+
+In most cases, building oneMKL is as simple as setting the compiler and
+selecting the desired backends to build with.
+
+On Linux (see `Building for Windows`_ for building on Windows):
+
+.. code-block:: bash
+
+  # Inside <path to onemkl>
+  mkdir build && cd build
+  cmake .. -DCMAKE_CXX_COMPILER=$CXX_COMPILER                       \ # Should be icpx or clang++
+          -DCMAKE_C_COMPILER=$C_COMPILER                            \ # Should be icx or clang
+          -DENABLE_MKLGPU_BACKEND=False                             \ # Optional: The MKLCPU backend is True by default.
+          -DENABLE_MKLGPU_BACKEND=False                             \ # Optional: The MKLGPU backend is True by default.
+          -DENABLE_<BACKEND_NAME>_BACKEND=True                      \ # Enable any other backend(s) (optional)
+          -DENABLE_<BACKEND_NAME_2>_BACKEND=True                    \ # Multiple backends can be enabled at once.
+          -DBUILD_FUNCTIONAL_TESTS=False                            \ # See page *Building and Running Tests* for more on building tests. ON by default.
+          -DBUILD_EXAMPLES=False                                      # Optional: On by default.
```

Comment:
There are some `ON` left in the comment.

Reply:
I've updated this, and hopefully all the other straggling ONs/OFFs in [8f4fc93](https://github.com/oneapi-src/oneMKL/pull/510/commits/8f4fc934065a380cae8462b00998e8f08866ba06).